pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -666,6 +669,10 @@ func (s *DiscoveryServer) DeltaAggregatedResources(stream ads.AggregatedDiscover
 func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) error {
 	// TODO: update the service deps based on NetworkScope
 
+	if err := con.modelNode.SetWorkloadLabels(s.Env); err != nil {
+		return err
+	}
```

Comment:
why do you need this during the push?

Reply:
You mean it should go low the check for the EDS-only update?
Makes sense.